src/main/java/pl/touk/sputnik/connector/stash/StashFacade.java

```diff
@@ -133,12 +126,10 @@ private FileComment toFileComment(String key, pl.touk.sputnik.review.Comment com
         return fileComment;
     }
 
-    private <T> List<T> transform(List<JSONObject> jsonList, Class<T> someClass) {
-        List<T> result = Lists.newArrayList();
+    private <T> List<T> transform(JSONArray jsonList, Class<T> someClass) {
+        List<T> result;
         try {
-            for (JSONObject jsonObject : jsonList) {
-                result.add(objectMapper.readValue(jsonObject.toJSONString(), someClass));
-            }
+            result = objectMapper.readValue(jsonList.toJSONString(), objectMapper.getTypeFactory().constructCollectionType(List.class, someClass));
```

Comment:
if you extract try/catch as method then you won't need result variable declaration without assigment

Reply:
Done